src/platform_linux/platform.c

```diff
@@ -356,8 +357,15 @@ platform_assert_msg(platform_log_handle *log_handle,
                     const char          *message,
                     va_list              varargs)
 {
-   static char assert_msg_fmt[] = "Assertion failed at %s:%d:%s(): \"%s\". ";
-   platform_log(
-      log_handle, assert_msg_fmt, filename, linenumber, functionname, expr);
+   static char assert_msg_fmt[] =
+      "OS-pid=%d, Thread-ID=%lu, Assertion failed at %s:%d:%s(): \"%s\". ";
+   platform_log(log_handle,
+                assert_msg_fmt,
+                getpid(),
```

Comment:
Maybe use `gettid()` instead?

https://man7.org/linux/man-pages/man2/gettid.2.html

> In a single-threaded process, the thread ID is equal to the process ID (PID, as returned by `getpid(2)`).  In a multithreaded process, all threads have the same PID, but each one has a unique TID.

I'd say that's strictly more information, and it makes it useful even for multi-threaded-single-process applications too.   It also better matches the output of tools I'm familiar with, e.g. `htop` will print the `tid`.

In which case, maybe update the format string to say `OS-tid` and `Splinter-tid` or something like that...?

Reply:
@rosenhouse - I am fine extending this to also report `gettid()` but think that OS-pid is also useful on its own. (Even if it will be the same as gettid for the single-threaded process case.)

The reason I say this is that in core files produced, OS-pid is registered. And I think it's expected that we crack open core files using, something like, `gdb core -p <pid>`.

Also, if the same field is sometimes `pid` and sometimes `gettid`, then it will be confusing to remember which case we are dealing with.

I'd prefer 'Thread-ID' for the splinter-tid, as that is what it is referred to consistently inside our code base.

How about we settle on:

`OS-pid=%d, OS-tid=%lu, Thread-ID=%lu, Assertion ...`

That should satisfy all users / usages.

Reply:
sounds good

Reply:
@rosenhouse -- I made this change we agreed to.

I did not add any new test to verify that the new `tid` value for a thread will be different than the one for the main program. That seemed a bit overboard and unnecessary.

CI is running now ... If you can clear this review, I'll merge this when CI passes.